myhpi/core/wagtail_hooks.py

```diff
@@ -37,7 +37,8 @@ def check_view_permissions(page, request, serve_args, serve_kwargs):
             target_groups = Group.objects.filter(
                 Q(name=request.user._ip_range_group_name) | Q(id__in=request.user.groups.all())
             )
-        if not any(group in page.visible_for.all() for group in target_groups):
+        matchesGroup = any(group in page.visible_for.all() for group in target_groups)
+        if not (matchesGroup or page.is_public):
```

Comment:
Renaming `matchesGroup` to `is_matching_group` would be more pythonic, I think

Reply:
has been renamed